jtravis/src/main/java/fr/inria/spirals/jtravis/helpers/BuildHelper.java

```diff
@@ -52,42 +56,126 @@ protected static BuildHelper getInstance() {
-
-            JsonObject commitJSON = allAnswer.getAsJsonObject("commit");
-            Commit commit = CommitHelper.getCommitFromJsonElement(commitJSON);
-            build.setCommit(commit);
+                build = createGson().fromJson(allAnswer, Build.class);
+                build.setCommitId(build.getCommit().getId());
+                build.setRepositoryId(build.getRepository().getId());
 
-            if (parentRepo != null) {
-                build.setRepository(parentRepo);
-            }
+                List<Integer> jobsId = new ArrayList<Integer>();
+                for (Job jobs : build.getJobs()) {
+                    jobsId.add(jobs.getId());
+                }
+                build.setJobIds(jobsId);
 
-            JsonObject configJSON = buildJSON.getAsJsonObject("config");
-            Config config = ConfigHelper.getConfigFromJsonElement(configJSON);
-            build.setConfig(config);
+            } else {
+                JsonObject buildJSON = allAnswer.getAsJsonObject("build");
+                build = createGson().fromJson(buildJSON, Build.class);
 
-            JsonArray arrayJobs = allAnswer.getAsJsonArray("jobs");
+                JsonObject commitJSON = allAnswer.getAsJsonObject("commit");
+                Commit commit = CommitHelper.getCommitFromJsonElement(commitJSON);
+                build.setCommit(commit);
 
-            for (JsonElement jobJSONElement : arrayJobs) {
-                Job job = JobHelper.createJobFromJsonElement((JsonObject)jobJSONElement);
-                build.addJob(job);
-            }
+                if (parentRepo != null) {
+                    build.setRepository(parentRepo);
+                }
 
+                JsonObject configJSON = buildJSON.getAsJsonObject("config");
+                Config config = ConfigHelper.getConfigFromJsonElement(configJSON);
+                build.setConfig(config);
 
+                JsonArray arrayJobs = allAnswer.getAsJsonArray("jobs");
 
+                for (JsonElement jobJSONElement : arrayJobs) {
+                    Job job = JobHelper.createJobFromJsonElement((JsonObject) jobJSONElement);
+                    build.addJob(job);
+                }
+            }
             return build;
         } catch (IOException e) {
             getInstance().getLogger().warn("Error when getting build id "+id+" : "+e.getMessage());
             return null;
         }
+
+    }
+
+    //@param offset set value to 0 for first request
+    private static void getBuildsFromSlugV3(String slug, List<Build> result, String QueryParameters, int offset, Date younger, Date older, int buildAfter, int buildBefore) {
+        slug = slug .replace("/","%2F");
+        String ENDPOINTV3 = "repo/"+slug+"/builds";
+        String resourceUrl = getInstance().getEndpoint()+ENDPOINTV3+"?offset="+offset;
+        Boolean byDate = false;
+        Boolean byNumber = false;
+        if(QueryParameters != "")
+            resourceUrl += "&"+QueryParameters;
```

Comment:
This param does not follow Java rules for naming parameters: you should not have a caps for starting